package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1231,22 +1266,78 @@ def print(self, verbose, *args):
         elif verbose:
             print(args)
 
-    def run(self, tau_max=20, start=0, stop=None, step=1, verbose=False):
+    def _correct_intermittency(self, intermittency, selected_ids):
         """
-        Computes and returns the survival probability timeseries
+        Pre-process Consecutive Intermittency
+        :param intermittency: the max gap allowed and to be corrected
+        :param selected_ids: modifies the selecteded IDs in place by adding atoms which left for <= :param intermittency
+        """
+        self.print('Correcting the selected IDs for intermittancy (gaps). ')
+        if intermittency == 0:
+            return
+
+        # If the atom is absent for a number of frames equal or smaller
```

Comment:
Stuff like this should be in actual docs.

Reply:
Or more precisely: the comments here are very good, but the docstring should tell a user what the function does (what problem does it solve). The comments should explain the how – the ideas behind the algorithm.

Reply:
Good point. I've moved the actual documentation to the top and introduced minor corrections in the comments.